src/main/java/com/webrest/common/service/RoleService.java

```diff
@@ -109,7 +135,105 @@ public Role getOne(Long roleId) {
 		return roleRepository.getOne(roleId);
 	}
 
-	public List<Role> getRolesWithAuthorization(List<Long> roleIds) {
-		return roleRepository.rolesWithAuthorizationRoleIdIn(roleIds);
+	public Map<AuthorizedFeature, Set<AuthorizedAction>> getRolesWithAuthorization(List<Long> roleIds) throws JsonProcessingException {
+
+		List<String> roleIdStrings = roleIds.stream().map((roleId) -> roleId.toString()).collect(Collectors.toList());
+		List<String> cachedFeatureActions = stringRedisTemplate.<String, String>opsForHash().multiGet(ROLE_ACCESS_HASH_KEY,
+				roleIdStrings);
+
+		Map<Long, Map<AuthorizedFeature, Set<AuthorizedAction>>> roleWiseCachedFeatureActions = roleWiseFeatureAction(roleIds, cachedFeatureActions);
+		List<Long> cacheMissedRoleIds = calculateMissedCacheRoleIds(roleIds, roleWiseCachedFeatureActions);
+
+		// TODO: Use apache common utils
+		if (CollectionUtils.isEmpty(cacheMissedRoleIds) == false) {
+			Map<Long, Map<AuthorizedFeature, Set<AuthorizedAction>>> roleWiseFeatureActionsFromDB = getRolesAuthorizedFeatureActions(
+					cacheMissedRoleIds);
+			updateCache(roleWiseFeatureActionsFromDB);
+			roleWiseCachedFeatureActions.putAll(roleWiseFeatureActionsFromDB);
+		}
+
+		// return roleRepository.rolesWithAuthorizationRoleIdIn(roleIds);
+		return mergeFeatureActions(roleWiseCachedFeatureActions.values());
+	}
+
+	private Map<Long, Map<AuthorizedFeature, Set<AuthorizedAction>>> roleWiseFeatureAction(List<Long> roleIds,
+			List<String> cachedFeatureActions) throws JsonMappingException, JsonProcessingException {
+		Map<Long, Map<AuthorizedFeature, Set<AuthorizedAction>>> roleWiseCachedFeatureAction = new HashMap<>();
+		for (int index = 0; index < cachedFeatureActions.size(); index++) {
+			if (cachedFeatureActions.get(index) != null) {
+				Long roleId = roleIds.get(index);
+				String serializedFeatureAction = cachedFeatureActions.get(index);
+				Map<AuthorizedFeature, Set<AuthorizedAction>> featureActions = objectMapper.readValue(
+						serializedFeatureAction, new TypeReference<Map<AuthorizedFeature, Set<AuthorizedAction>>>() {
+
+						});
+
+				roleWiseCachedFeatureAction.put(roleId, featureActions);
+			}
+		}
+		return roleWiseCachedFeatureAction;
+	}
+
+	private List<Long> calculateMissedCacheRoleIds(List<Long> roleIds,
+			Map<Long, Map<AuthorizedFeature, Set<AuthorizedAction>>> roleWiseCachedFeatureActions) {
+		return roleIds.stream().filter(roleId -> {
+			return roleWiseCachedFeatureActions.containsKey(roleId) == false;
+		}).collect(Collectors.toList());
+	}
+
+	private Map<Long, Map<AuthorizedFeature, Set<AuthorizedAction>>> getRolesAuthorizedFeatureActions(List<Long> roleIds) {
```

Comment:
On the top similar function exists, Please rename this function, if it does something different.